staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -226,6 +272,60 @@ func RunTestWatchFromNoneZero(ctx context.Context, t *testing.T, store storage.I
 	testCheckResult(t, watch.Modified, w, out)
 }
 
+func RunTestDelayedWatchDelivery(ctx context.Context, t *testing.T, store storage.Interface) {
+	_, storedObj := testPropagateStore(ctx, t, store, &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test-ns"}})
+	startRV := storedObj.ResourceVersion
+
+	watcher, err := store.Watch(ctx, "/pods/test-ns", storage.ListOptions{ResourceVersion: startRV, Predicate: storage.Everything, Recursive: true})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Depending on the implementation, different number of events that
```

Comment:
Would it be useful to make these implementation-specific counts part of the `RunTest...` signature?

Reply:
That was the case for my original implementation. But for etcd this number doesn't really exist at all, as watches are independent on etcd.
So to make the tests as simple as possible (given that we don't have any reasonable number to put here for etcd), I would leave as is, until at least we have another implementation of storage interface.